docs/protocol.md

```diff
@@ -0,0 +1,364 @@
+# HTTP protocol for RAG chat apps
```

Comment:
I almost feel like there should be a separate repository documenting this protocol that has samples on how to integrate with it and stand up simple servers / clients. But for now I think this is OK

Reply:
I'm happy to stick this in a separate repository, I have a reputation for repository creation. However, the "context" properties should be documented here, since they could vary per server/client. Let me know!

Reply:
Made a new repo and made changes there, see: https://github.com/Azure-Samples/ai-chat-app-protocol/pull/1

Reply:
I'm going to remove the protocol document from here, since it's now in a separate repo. For this repo, my goal is to bring in Quart-schema as that can generate up-to-date documentation for anyone using the API.

Reply:
Quart-schema issue filed here: https://github.com/Azure-Samples/azure-search-openai-demo/issues/1248